### PR TITLE
refactor(types): move custom manager types 

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
distribute types CustomManager and RegexManager

## Changes
- Moved custom manager types inside the manager/custom folder so that it is easy to find and refactor them since they are clser to their sources.
- Updating interfaces `CustomManager` and `CustomExtractConfig` to extends custom manager configs

## Context
Part of: #19066 
- Make changes to CustomManager types to simplify adding new custom manager defns to it

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/custom/types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export interface CustomManager extends Partial<RegexManagerConfig> {
+  customType: CustomManagerName;
+  fileMatch: string[];
+}
+
+// NOTE:
+// the two interfaces might seem similar but they have different usage...
+// CustomManager interface consists of all fields that a custom manager can have
+// whereas
+// CustomExtractConfig consists of the fields that custom managers need when performing extraction
```
Comment (RahulGautamSingh) [inline lib/modules/manager/custom/types.ts:16]:
```suggestion
// CustomExtractConfig consists of the fields that custom managers need to extract deps from a package file
```

Comment (RahulGautamSingh) [inline lib/modules/manager/custom/types.ts:16]:
```suggestion
// the two interfaces might seem similar but they are different like ManagerConfig and ExtractConfig
```
Does this seem clearer than the explanation?

File: lib/modules/manager/custom/types.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  customType: CustomManagerName;
+  fileMatch: string[];
+}
+
+// NOTE:
+// the two interfaces might seem similar but they have different usage...
+// CustomManager interface consists of all fields that a custom manager can have
+// whereas
+// CustomExtractConfig consists of the fields that custom managers need to extract deps from a package file
+// ie. options necessary for the function extractPackageFile
```
Comment (RahulGautamSingh) [inline lib/modules/manager/custom/types.ts:17]:
```suggestion
// the two interfaces might seem similar but they have different usage similar to ManagerConfig and ExtractConfig
```

File: lib/modules/manager/custom/regex/strategies.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ function processRecursive(parameters: RecursionParameter): PackageDependency[] {
     regexes,
     config,
   }: RecursionParameter = parameters;
-  // abort if we have no matchString anymore
+  // abort if we have no matched strings anymore
```
Comment (viceice) [inline lib/modules/manager/custom/regex/strategies.ts:92]: I'm not sure to change this.
@rarkins @HonkingGoose WDYT?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/custom/regex/strategies.ts (hunk 3 above)
Comment (rarkins) [reply lib/modules/manager/custom/regex/strategies.ts:92]: No, revert

Comment (RahulGautamSingh) [inline lib/modules/manager/custom/regex/strategies.ts:92]:
```suggestion
  // abort if we have no matchStrings anymore
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**